### PR TITLE
Add support for RISC-V attributes

### PR DESCRIPTION
**Pull request by barbedo**
RISC-V ELF attributes follow the same structure as the ARM ones:
- https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/master/riscv-elf.adoc#attributes

This patch adds:
- RISC-V specific ELF definitions.
- Refactor the ARM attribute classes and use them to support RISC-V Attributes.
- A unit test and a sample ELF file compiled using [riscv-collab/riscv-gnu-toolchain](https://github.com/riscv-collab/riscv-gnu-toolchain) with an arbitrary ISA.
  - I've created `simple_gcc.riscv.c` file under `test/testfiles_for_unittests` just to document the source of the compiler and its command.

### CODE REVIEW

File: elftools/elf/sections.py [hunk 1]
```diff
@@ -298,68 +298,34 @@ def iter_stabs(self):
             self.stream.seek(offset)
             yield stabs
 
-
-class ARMAttribute(object):
-    """ ARM attribute object - representing a build attribute of ARM ELF files.
+class Attribute(object):
+    """ Attribute object - representing a build attribute of an ELF files.
```
Comment (eliben) [inline elftools/elf/sections.py:302]: s/an//

File: elftools/elf/sections.py [hunk 2]
```diff
@@ -500,8 +467,133 @@ def _make_subsections(self):
...
+
+            if self.tag != 'TAG_FILE':
+                self.extra = []
+                s_number = struct_parse(structs.Elf_uleb128('s_number'), stream)
+
+                while s_number != 0:
+                    self.extra.append(s_number)
+                    s_number = struct_parse(structs.Elf_uleb128('s_number'),
+                                            stream
+                               )
```
Comment (eliben) [inline elftools/elf/sections.py:495]: nit: formatting -- move this `)` to previous line

File: elftools/elf/sections.py [hunk 3]
```diff
@@ -500,8 +467,133 @@ def _make_subsections(self):
...
+
+class ARMAttributesSubsection(AttributesSubsection):
+    """ Subsection of an ELF .ARM.attributes section.
+    """
+    def __init__(self, stream, structs, offset):
+        super(ARMAttributesSubsection, self).__init__(
+            stream, structs, offset,
+            structs.Elf_Attr_Subsection_Header,
+            ARMAttributesSubsubsection
+        )
```
Comment (eliben) [inline elftools/elf/sections.py:536]: Same here: move `)` up

File: elftools/elf/sections.py [hunk 4]
```diff
@@ -500,8 +467,133 @@ def _make_subsections(self):
...
+
+            if self.tag != 'TAG_FILE':
+                self.extra = []
+                s_number = struct_parse(structs.Elf_uleb128('s_number'), stream)
+
+                while s_number != 0:
+                    self.extra.append(s_number)
+                    s_number = struct_parse(structs.Elf_uleb128('s_number'),
+                                            stream
+                                            )
```
Comment (eliben) [inline elftools/elf/sections.py:565]: here too -- `)` up

**Review by eliben — COMMENTED** [review]
Nice, thanks! Left some minor comments

File: elftools/elf/sections.py (hunk 1 above)
Comment (barbedo) [reply elftools/elf/sections.py]: Oops. Fixed.

File: elftools/elf/sections.py (hunk 2 above)
Comment (barbedo) [reply elftools/elf/sections.py]: Done.

File: elftools/elf/sections.py (hunk 3 above)
Comment (barbedo) [reply elftools/elf/sections.py]: Done.

File: elftools/elf/sections.py (hunk 4 above)
Comment (barbedo) [reply elftools/elf/sections.py]: And done. I also changed two other occurrences that were already there.

**Review by eliben — APPROVED** [review]

**Pull request merged**